backend/services/implementations/statisticService.ts

```diff
@@ -74,6 +75,25 @@ class StatisticService implements IStatisticService {
     return aggCursor[0]?.numSubmittedTests ?? 0;
   }
 
+  async getMeanScoreByTest(testId: string): Promise<number> {
+    const pipeline = [
+      filterTestsByTestId(testId),
+      {
+        $project: {
+          results: filterUngradedTests,
+        },
+      },
+      unwindResults,
+      sumTestSubmissions,
+    ];
+
+    const aggCursor = await MgTestSession.aggregate(pipeline);
+    const sum = aggCursor[0]?.totalScore ?? 0;
+    const count = await this.getSubmissionCountByTest(testId);
```

Comment:
instead of calling this function and having to make another aggregation, can we directly compute the mean in the pipeline? not sure if [$avg](https://www.mongodb.com/docs/manual/reference/operator/aggregation/avg/) will be of use (or you might need to do some extra preprocessing) but you can take a look into it!

Reply:
Ahh I was considering that, the ticket mentioned calling `getSubmissionCountByTest` and dividing manually so that's what I did, but I'll look into $avg

Reply:
Changed!

Reply:
ahh sorry for the miscommunication